### ISSUE TITLE
[firebase_messaging]: onBackgroundMessage handler not triggering for iOS

### ISSUE THREAD

**Issue by RyanMannix** [firebase/flutterfire#11286]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Are you aware of the differences between iOS and Android background message handling?

- [X] I understand that iOS and Android background messages behave differently, and I've designed my application with that in mind.

### Do you have an active Apple Developer account?

- [X] I have an active Apple Developer account.

### Are you using a physical iOS device to test background messages?

- [X] I am using a physical iOS device to test background messages.

### Have you enabled "Remote Notifications" & "Background Mode" (Checking options for "Background Processing" & "Remote Notifications") in your app's Xcode project?

<img width="521" alt="Screenshot 2023-07-14 at 1 56 23 PM" src="https://github.com/firebase/flutterfire/assets/50586508/521bf108-16aa-4c42-aca2-f042e313369e">

### Have you created an APNs key in your Apple Developer account & uploaded this APNs key to your Firebase console?

![Screenshot 2023-07-14 at 1 57 59 PM](https://github.com/firebase/flutterfire/assets/50586508/6bbfc98c-a3f8-4fae-a42f-fd3baba86fc0)

### Have you disabled method swizzling for Firebase in your app?

<img width="353" alt="Screenshot 2023-07-14 at 2 02 31 PM" src="https://github.com/firebase/flutterfire/assets/50586508/0d88e457-303e-44e3-9b99-ccff7e709f82">

### Are you sending messages to your app from the Firebase Admin SDK?

	client, err := app.App.Messaging(ctx)
	if err != nil {
		log.Fatalf("error getting Messaging client: %v\n", err)
	}

	message := &messaging.MulticastMessage{
		Notification: &messaging.Notification{
			Title: n.Title,
			Body:  n.Body,
		},

		Data: n.Payload,

		Android: &messaging.AndroidConfig{
			Notification: &messaging.AndroidNotification{
				ChannelID: "high_importance_channel",
				Priority:  messaging.PriorityHigh,
			},
		},

		APNS: &messaging.APNSConfig{
			Payload: &messaging.APNSPayload{
				Aps: &messaging.Aps{
					Sound:            "default",
					ContentAvailable: true,
				},
			},
		},

		Tokens: n.FcmTokens,
	}

	batchResponse, err := client.SendMulticast(ctx, message)
	if err != nil {
		log.Printf("Error sending message to firebase: %s", err)
		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
		return
	}

### Have you requested permission from the user to receive notifications?

- [X] I have the relevant permission to receive notifications.

### Have you used the 'Console' application on your macOS device to check if the iOS device's system is throttling your background messages?

`default	14:12:30.157126-0400	SpringBoard	SBIconView touches began with event: <UITouchesEvent: 0x283491e10> timestamp: 698243 touches: {(
    <UITouch: 0x99eabda40> phase: Began tap count: 1 force: 0.000 window: <SBHomeScreenWindow: 0x99a9f9fd0; HomeScreen-0x99a9f9fd0-5; baseClass = UIWindow; frame = (0 0; 390 844); clipsToBounds = YES; opaque = NO; gestureRecognizers = <NSArray: 0x2801751a0>; layer = <UIWindowLayer: 0x2801759e0>> responder: <SBIconView: 0x99b93e200; frame: {{300, 469}, {60, 74}}; icon: <SBApplicationIcon: 0x282289c00; nodeID: <my app bundle>.dev; bundleID: <my app bundle>.dev>; location: SBIconLocationRoot; isTouchDownInIcon: YES> location in window: {335.33332824707031, 509.66665649414062} previous location in window: {335.33332824707031, 509.66665649414062} location in view: {35.333328247070312, 40.666656494140625} previous location in view: {35.333328247070312, 40.666656494140625}
)}, tap gesture: <SBIconTapGestureRecognizer: 0x99eae4c40; baseClass = UITapGestureRecognizer; state = Possible; view = <SBIconView: 0x99b93e200>; target= <(action=tapGestu
default	14:12:31.394785-0400	runningboardd	Assertions for process will expire soon: [xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403]
default	14:12:31.394849-0400	runningboardd	Assertions for process will expire soon: [xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404]
default	14:12:31.395143-0400	runningboardd	Acquiring assertion targeting [xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] from originator [osservice<com.apple.runningboardd>:81318] with description <RBSAssertionDescriptor| "Will expire assertions soon" ID:81318-81318-83013 target:90403 attributes:[
	<RBSCPUAccessGrant| role:NonUserInteractive>,
	<RBSDurationAttribute| invalidationDuration:4.63 warningDuration:0.00 startPolicy:Delayed-Fixed endPolicy:Invalidate>,
	<RBSRunningReasonAttribute| runningReason:10101>
	]>
default	14:12:31.395407-0400	runningboardd	Assertion 81318-81318-83013 (target:[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403]) will be created as inactive as start-time-defining assertions exist
default	14:12:31.398022-0400	runningboardd	Assertions for process will expire soon: [xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403]
default	14:12:33.841988-0400	SpringBoard	Received incoming message on topic <my app bundle>.dev at priority 10
default	14:12:33.857438-0400	SpringBoard	[<my app bundle>.dev] Received remote notification request 6BDF-6943 [ waking: 0, hasAlertContent: 1, hasSound: 1 hasBadge: 0 hasContentAvailable: 0 hasMutableContent: 0 pushType: Alert]
default	14:12:33.857502-0400	SpringBoard	[<my app bundle>.dev] Process delivery of push notification 6BDF-6943
default	14:12:33.857707-0400	SpringBoard	[<my app bundle>.dev] Badge can be set for notification 6BDF-6943: 0 [ canBadge: 1 badgeNumber: (null) ]
default	14:12:33.857789-0400	SpringBoard	Getting effectiveSectionInfo for section identifier: <my app bundle>.dev
default	14:12:33.857896-0400	SpringBoard	[<my app bundle>.dev] Getting effective section info
default	14:12:33.859149-0400	SpringBoard	[<my app bundle>.dev] Got effective section info [ hasResult: 1 ]
default	14:12:33.859297-0400	SpringBoard	Getting effectiveSectionInfo for section identifier: <my app bundle>.dev
default	14:12:33.859363-0400	SpringBoard	[<my app bundle>.dev] Getting effective section info
default	14:12:33.860447-0400	SpringBoard	[<my app bundle>.dev] Got effective section info [ hasResult: 1 ]
default	14:12:33.860569-0400	SpringBoard	[<my app bundle>.dev] Saving notification 6BDF-6943: YES [ hasAlertContent: YES, shouldPresentAlert: YES settingsShouldSave: YES]
default	14:12:33.863645-0400	SpringBoard	[<my app bundle>.dev] Delivered user visible push notification 6BDF-6943
default	14:12:33.864379-0400	SpringBoard	[<my app bundle>.dev] Load 0 pending notification dictionaries
default	14:12:33.864583-0400	SpringBoard	[<my app bundle>.dev] Adding notification 6BDF-6943 [ hasAlertContent: 1, shouldPresentAlert: 1 hasSound: 1 shouldPlaySound: 1 ]; interruption-level: 1; destinations 398: (
    NotificationCenter,
    LockScreen,
    Alert,
    Spoken,
    Forwarding
)
default	14:12:33.865118-0400	SpringBoard	BBDataProviderProxy <my app bundle>.dev has enqueued a bulletin request
default	14:12:33.865161-0400	SpringBoard	BBDataProviderProxy <my app bundle>.dev is now sending enqueued bulletin request to BBServer
default	14:12:33.867218-0400	SpringBoard	Publishing bulletin for section <<my app bundle>.dev>: subsectionIDs: (null), bulletinID = <39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF>, expiration date = <Fri Jul 21 14:12:33 2023>, expiration events <0>
default	14:12:33.868513-0400	SpringBoard	Attempt gateway BBObserverGatewayLocalName for bulletin 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev
default	14:12:33.868578-0400	SpringBoard	Gateway BBObserverGatewayLocalName sending add bulletin lights and sirens YES for 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev with timeout 20.000000 WITH handler
default	14:12:33.868617-0400	SpringBoard	Gateway BBObserverGatewayLocalName add timer (Fri Jul 14 14:12:53 2023) for bulletin 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev
default	14:12:33.868676-0400	SpringBoard	Gateway BBObserverGatewayLocalName start timer (Fri Jul 14 14:12:53 2023) for out for bulletin 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev
default	14:12:33.870784-0400	SpringBoard	<private> adding bulletin 6BDF-6943 for feed 3131 in section <my app bundle>.dev playLightsAndSirens: YES
default	14:12:33.871212-0400	SpringBoard	Gateway BBObserverGatewayLocalName received DID play lights and sirens for bulletin 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev
default	14:12:33.871265-0400	SpringBoard	Gateway BBObserverGatewayLocalName invalidated timer (Fri Jul 14 14:12:53 2023) for out for bulletin 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev
default	14:12:33.872692-0400	SpringBoard	Gateway enumerator received from BBObserverGatewayLocalName DID play lights and sirens for 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev currently NOT cancelled
default	14:12:33.872760-0400	SpringBoard	Attempt gateway GizmoGateway for bulletin 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev
default	14:12:33.872838-0400	SpringBoard	Gateway GizmoGateway sending add bulletin lights and sirens NO for 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev with NO timeout and NO handler
default	14:12:33.872889-0400	SpringBoard	Gateway enumerator received from GizmoGateway DID NOT play lights and sirens for 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev currently NOT cancelled
default	14:12:33.872914-0400	SpringBoard	Gateway enumerator was completed for bulletin 39EC3CDA-F48B-4BF4-B8AD-7939A6C553AF in section <my app bundle>.dev
default	14:12:33.875401-0400	SpringBoard	NCBulletinNotificationSource adding bulletin 6BDF-6943 for feed 3131 in section <my app bundle>.dev
default	14:12:33.901983-0400	SpringBoard	[com.apple.springboard.NCBulletinNotificationSource:07C63984-A7F1-4F68-A2B6-C6867435F902] Resolving behavior for event, details=<DNDMutableClientEventDetails: 0x282ed92c0; identifier: 'D530998C-DC05-47E6-A238-5EA3AB4DFBC5'; bundleIdentifier:: <my app bundle>.dev; type: Default; urgency: Default; sender: (null); threadIdentifier: EB93D4264C930BB3; filterCritera: (null); notifyAnyway: 0; behavior: Default>
default	14:12:33.903645-0400	donotdisturbd	Resolving breakthrough through for configuration: <DNDMutableConfiguration: 0xa10e20b90; suppressionType: delay delivery; allowedApplicationIdentifiers: {
    "<DNDApplicationIdentifier: 0xa10d0fe20; bundleID: com.kbi.timecardFlutter; platform: iOS>" = "<DNDMutableApplicationConfiguration: 0xa10e0e330; minimumBreakthroughUrgency: all; allowedThreads: {(\n)}; deniedThreads: {(\n)}; >";
    "<DNDApplicationIdentifier: 0xa10d29e40; bundleID: <my app bundle>.dev; platform: iOS>" = "<DNDMutableApplicationConfiguration: 0xa10e20570; minimumBreakthroughUrgency: all; allowedThreads: {(\n)}; deniedThreads: {(\n)}; >";
    "<DNDApplicationIdentifier: 0xa10d29ea0; bundleID: <my app bundle>.beta; platform: iOS>" = "<DNDMutableApplicationConfiguration: 0xa10e18c60; minimumBreakthroughUrgency: all; allowedThreads: {(\n)}; deniedThreads: {(\n)}; >";
    "<DNDApplicationIdentifier: 0xa10d1db60; bundleID: <my app bundle>; platform: iOS>" = "<DNDMutableApplicationConfiguration: 0xa10e209a0; minimumBreakthroughUrgency: all; allowedThreads: {(\n)}; with event details application: <decode: missing data>, sender: <decode: missing data>, urgency: <decode: missing data>, eventType: <decode: missing data>, threadIdentifier: <decode: missing data>
default	14:12:33.906888-0400	donotdisturbd	Breakthrough is allowed for configuration <DNDMutableConfiguration: 0xa10e20b90; suppressionType: delay delivery; allowedApplicationIdentifiers: {
    "<DNDApplicationIdentifier: 0xa10d0fe20; bundleID: com.kbi.timecardFlutter; platform: iOS>" = "<DNDMutableApplicationConfiguration: 0xa10e0e330; minimumBreakthroughUrgency: all; allowedThreads: {(\n)}; deniedThreads: {(\n)}; >";
    "<DNDApplicationIdentifier: 0xa10d29e40; bundleID: <my app bundle>.dev; platform: iOS>" = "<DNDMutableApplicationConfiguration: 0xa10e20570; minimumBreakthroughUrgency: all; allowedThreads: {(\n)}; deniedThreads: {(\n)}; >";
    "<DNDApplicationIdentifier: 0xa10d29ea0; bundleID: <my app bundle>.beta; platform: iOS>" = "<DNDMutableApplicationConfiguration: 0xa10e18c60; minimumBreakthroughUrgency: all; allowedThreads: {(\n)}; deniedThreads: {(\n)}; >";
    "<DNDApplicationIdentifier: 0xa10d1db60; bundleID: <my app bundle>; platform: iOS>" = "<DNDMutableApplicationConfiguration: 0xa10e209a0; minimumBreakthroughUrgency: all; allowedThreads with event details: <decode: missing data>.
default	14:12:33.906958-0400	donotdisturbd	Breakthrough is NOT allowed for global settings with event details: <DNDMutableClientEventDetails: 0xa109544a0; identifier: 'D530998C-DC05-47E6-A238-5EA3AB4DFBC5'; bundleIdentifier:: <my app bundle>.dev; type: Default; urgency: Default; sender: (null); threadIdentifier: EB93D4264C930BB3; filterCritera: (null); notifyAnyway: 0; behavior: Default>.
default	14:12:33.907074-0400	donotdisturbd	Event was resolved: resolution=<DNDSEventBehaviorResolution: 0xa10a41b40; UUID: 6ABC3C7E-D192-42F0-BDF0-4B7031596791; date: 2023-07-14 18:12:33 +0000; eventBehavior: <DNDClientEventBehavior: 0xa10a4cfa0; eventDetails: <DNDClientEventDetails: 0xa10a4c060; identifier: 'D530998C-DC05-47E6-A238-5EA3AB4DFBC5'; bundleIdentifier:: <my app bundle>.dev; type: Default; urgency: Default; sender: (null); threadIdentifier: EB93D4264C930BB3; filterCritera: (null); notifyAnyway: 0; behavior: Default>; interruptionSuppression: none; resolutionReason: mode configuration for application; activeModeUUID: 09CC4F96-1817-4E08-AC34-583E81C254D7>; clientIdentifier: 'com.apple.springboard.NCBulletinNotificationSource'; outcome: allowed; reason: mode configuration for application>
default	14:12:33.907474-0400	SpringBoard	[com.apple.springboard.NCBulletinNotificationSource:07C63984-A7F1-4F68-A2B6-C6867435F902] Resolved event, details=<DNDMutableClientEventDetails: 0x282ed92c0; identifier: 'D530998C-DC05-47E6-A238-5EA3AB4DFBC5'; bundleIdentifier:: <my app bundle>.dev; type: Default; urgency: Default; sender: (null); threadIdentifier: EB93D4264C930BB3; filterCritera: (null); notifyAnyway: 0; behavior: Default> behavior=<DNDClientEventBehavior: 0x281f926a0; eventDetails: <DNDClientEventDetails: 0x282edb340; identifier: 'D530998C-DC05-47E6-A238-5EA3AB4DFBC5'; bundleIdentifier:: <my app bundle>.dev; type: Default; urgency: Default; sender: (null); threadIdentifier: EB93D4264C930BB3; filterCritera: (null); notifyAnyway: 0; behavior: Default>; interruptionSuppression: none; resolutionReason: mode configuration for application; activeModeUUID: 09CC4F96-1817-4E08-AC34-583E81C254D7>
default	14:12:33.907855-0400	SpringBoard	Posting notification id: 6BDF-6943; section: <my app bundle>.dev; thread: 1C84-85AA; category: ; timestamp: 2023-07-14 18:12:33 +0000; interruption-level: active; relevance-score: 0.00; filter-criteria: (null); actions: [ minimal: 0 (0 text), default: 0 (0 text) ]; destinations: [ {(
    BulletinDestinationCoverSheet,
    BulletinDestinationBanner,
    BulletinDestinationNotificationCenter,
    BulletinDestinationLockScreen
)} ]
default	14:12:33.908787-0400	SpringBoard	Incoming Section [NCNotificationStructuredSectionList] inserting notification request 6BDF-6943 in new group Group List [<my app bundle>.dev:1C84-85AA]
default	14:12:33.908828-0400	SpringBoard	Group List [<my app bundle>.dev:1C84-85AA] inserting notification request 6BDF-6943 at index 0
default	14:12:36.650825-0400	locationd	{"msg":"#CLIUA Marking change", "clientKey":"<my app bundle>.dev", "reason":"Decaying in-use status from process state", "assertionLevel":4, "coming":0}
default	14:12:36.651102-0400	locationd	{"msg":"#CLIUA updating AssertionRecord", "ClientKey":"<my app bundle>.dev", "AssertionLevel":{"type":"decode failure","raw value":4,"expected type":"Generic"}}
default	14:12:36.651281-0400	locationd	{"msg":"#CLIUA AssertionRecord updated", "ClientKey":"<my app bundle>.dev", "AssertionLevel":{"type":"decode failure","raw value":0,"expected type":"Generic"}}
default	14:12:36.651417-0400	locationd	{"msg":"#CLIUA in-use level changed for client", "ClientKey":"<my app bundle>.dev"}
default	14:12:36.653250-0400	locationd	{"msg":"computing freshAuthorizationContext", "ClientKey":"<my app bundle>.dev", "BigSwitch":1, "InUseLevel":{"type":"decode failure","raw value":0,"expected type":"Generic"}}
default	14:12:37.125860-0400	runningboardd	Assertion did invalidate due to timeout: 81318-90397-82987 (target:[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403])
default	14:12:37.125891-0400	runningboardd	Assertion did invalidate due to timeout: 81318-90397-82988 (target:[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404])
default	14:12:37.327305-0400	runningboardd	Removed last relative-start-date-defining assertion for process xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404
default	14:12:37.327370-0400	runningboardd	Removed last relative-start-date-defining assertion for process xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403
default	14:12:37.327394-0400	runningboardd	Activating delayed assertion 81318-81318-83013 (target:[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403])
default	14:12:37.328246-0400	runningboardd	Assertions for process will expire soon: [xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403]
default	14:12:37.328356-0400	runningboardd	Calculated state for xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403: running-active (role: NonUserInteractive)
default	14:12:37.334175-0400	runningboardd	Calculated state for xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404: running-suspended (role: None)
default	14:12:37.346002-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] Set jetsam priority to 0 [0] flag[1]
default	14:12:37.348278-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] Set darwin role to: NonUserInteractive
default	14:12:37.348484-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] set Memory Limits to Hard Inactive (840)
default	14:12:37.348768-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] Set jetsam priority to 0 [0] flag[1]
default	14:12:37.349071-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] Suspending task.
default	14:12:37.350472-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] Shutdown sockets (SVC)
default	14:12:37.350573-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] Set darwin role to: None
default	14:12:37.350666-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] set Memory Limits to Hard Inactive (300)
default	14:12:37.351437-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] check if suspended process is holding locks
error	14:12:37.352787-0400	runningboardd	[application<<my app bundle>.dev>:90397] client not entitled to get limitationsForInstance: <Error Domain=RBSServiceErrorDomain Code=1 "Client not entitled" UserInfo={RBSEntitlement=com.apple.runningboard.process-state, NSLocalizedFailureReason=Client not entitled, RBSPermanent=false}>
default	14:12:37.354656-0400	runningboardd	Acquiring assertion targeting [xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] from originator [application<<my app bundle>.dev>:90397] with description <RBSAssertionDescriptor| "NetworkProcess Background Assertion" ID:81318-90397-83015 target:90403 attributes:[
	<RBSDomainAttribute| domain:"com.apple.webkit" name:"Background" sourceEnvironment:"(null)">
	]>
default	14:12:37.354859-0400	runningboardd	Assertion 81318-90397-83015 (target:[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403]) will be created as active
default	14:12:37.355906-0400	runningboardd	Deactivating delayed assertion 81318-81318-83013 (target:[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403])
error	14:12:37.355981-0400	runningboardd	[application<<my app bundle>.dev>:90397] client not entitled to get limitationsForInstance: <Error Domain=RBSServiceErrorDomain Code=1 "Client not entitled" UserInfo={RBSEntitlement=com.apple.runningboard.process-state, NSLocalizedFailureReason=Client not entitled, RBSPermanent=false}>
default	14:12:37.356195-0400	runningboardd	Calculated state for xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403: running-active (role: Background)
default	14:12:37.356329-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] Set jetsam priority to 40 [0] flag[1]
default	14:12:37.356379-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] Set darwin role to: Background
default	14:12:37.356427-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] set Memory Limits to Hard Inactive (840)
default	14:12:37.356885-0400	runningboardd	Acquiring assertion targeting [xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] from originator [application<<my app bundle>.dev>:90397] with description <RBSAssertionDescriptor| "GPUProcess Background Assertion" ID:81318-90397-83016 target:90404 attributes:[
	<RBSDomainAttribute| domain:"com.apple.webkit" name:"Background" sourceEnvironment:"(null)">
	]>
default	14:12:37.357089-0400	runningboardd	Assertion 81318-90397-83016 (target:[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404]) will be created as active
default	14:12:37.358280-0400	runningboardd	Calculated state for xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404: running-active (role: Background)
default	14:12:37.358689-0400	runningboardd	Acquiring assertion targeting [xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] from originator [application<<my app bundle>.dev>:90397] with description <RBSAssertionDescriptor| "NetworkProcess Suspended Assertion" ID:81318-90397-83017 target:90403 attributes:[
	<RBSDomainAttribute| domain:"com.apple.webkit" name:"Suspended" sourceEnvironment:"(null)">
	]>
default	14:12:37.358738-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] Set jetsam priority to 40 [0] flag[1]
default	14:12:37.358840-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] Resuming task.
default	14:12:37.358896-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] Set darwin role to: Background
default	14:12:37.358932-0400	runningboardd	Assertion 81318-90397-83017 (target:[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403]) will be created as active
default	14:12:37.358985-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] set Memory Limits to Hard Inactive (300)
default	14:12:37.359546-0400	runningboardd	Calculated state for xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403: running-active (role: Background)
default	14:12:37.359759-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] set Memory Limits to Hard Inactive (840)
default	14:12:37.365642-0400	runningboardd	Invalidating assertion 81318-90397-83015 (target:[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403]) from originator [application<<my app bundle>.dev>:90397]
default	14:12:37.371031-0400	runningboardd	Invalidating assertion 81318-90397-83016 (target:[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404]) from originator [application<<my app bundle>.dev>:90397]
default	14:12:37.372864-0400	runningboardd	[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404] termination reported by launchd (2, 9, 9)
default	14:12:37.373652-0400	runningboardd	Removing process: [xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404]
default	14:12:37.373702-0400	runningboardd	removeJobWithInstance called for identity without existing job [xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404]
default	14:12:37.373750-0400	runningboardd	Removing assertions for terminated process: [xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404]
default	14:12:37.373847-0400	runningboardd	Removed last relative-start-date-defining assertion for process xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404
default	14:12:37.375824-0400	runningboardd	Calculated state for xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404: none (role: None)
default	14:12:37.375974-0400	runningboardd	Calculated state for xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404: none (role: None)
default	14:12:37.387746-0400	runningboardd	Invalidating assertion 81318-90397-82892 (target:[application<<my app bundle>.dev>:90397]) from originator [application<<my app bundle>.dev>:90397]
error	14:12:37.468786-0400	runningboardd	RBSStateCapture remove item called for untracked item 81318-90397-83016 (target:[xpcservice<com.apple.WebKit.GPU([application<<my app bundle>.dev>:90397])>:90404:90404])
default	14:12:37.469417-0400	runningboardd	Calculated state for application<<my app bundle>.dev>: running-active (role: Background)
default	14:12:37.469496-0400	runningboardd	[application<<my app bundle>.dev>:90397] set Memory Limits to Hard Inactive (3072)
default	14:12:37.470146-0400	runningboardd	Attempting to rename power assertion 38405 for target application<<my app bundle>.dev> to application<<my app bundle>.dev>81318-90397-83002:Shared Background Assertion 25 for <my app bundle>.dev(FinishTask)
default	14:12:37.470319-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] Set jetsam priority to 0 [0] flag[1]
default	14:12:37.470367-0400	runningboardd	[xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403:90403] set Memory Limits to Hard Inactive (840)
default	14:12:37.470400-0400	runningboardd	Calculated state for xpcservice<com.apple.WebKit.Networking([application<<my app bundle>.dev>:90397])>:90403: running-active (role: Background)
`

### Additional context and comments

Please let me know if there's anything else I should provide.

I've tested on a wide array of simulators/iOS versions, and the hard phone was an iphone 13Pro iOS 16.5.1

flutter firebase_messaging 14.6.5
fcm server using golang firebase admin SDK  firebase.google.com/go/v4 v4.11.0

The actual issue:
Both android and ios receive notifications both in foreground and background. The notification is displayed just fine.

In the flutter app, the `onBackgroundMessage(firebaseMessagingBackgroundHandler)` never gets triggered (nothing in `firebaseMessagingBackgroundHandler` ever happens.

```
Future<void> main() async {
  WidgetsFlutterBinding.ensureInitialized();
  await dotenv.load(fileName: 'lib/.env');


  if (Platform.isAndroid) {
    await AndroidInAppWebViewController.setWebContentsDebuggingEnabled(true);
  }

  WidgetsFlutterBinding.ensureInitialized();
  await Firebase.initializeApp(options: DevFirebaseOptions.currentPlatform);

  if (!kIsWeb) {
    // handles runtime permissions
    await requestNotificationPermissions();
    // Ensures local notification file exists
    await setupLocalNotificationsFile();
    // handles background notifications
    FirebaseMessaging.onBackgroundMessage(firebaseMessagingBackgroundHandler);
  }
.../...

Future<void> firebaseMessagingBackgroundHandler(RemoteMessage message) async {
  await Firebase.initializeApp(options: DevFirebaseOptions.currentPlatform);
  debugPrint('Got a message whilst in the background!'); // never gets logged
  // ... additional logic that never happens
```

**Comment by darshankawar** [comment]
@RyanMannix 
Check below similar issues and see if they help in resolving your case or not.

https://github.com/firebase/flutterfire/issues/10082
https://github.com/firebase/flutterfire/issues/10603

**Comment by RyanMannix** [comment]
@darshankawar I will be able to test this week and provide some details about the results for those specific issues

**Comment by RyanMannix** [comment]
The golang firebase sdk message config that works is this:
```
APNS: &messaging.APNSConfig{
	Payload: &messaging.APNSPayload{
		Aps: &messaging.Aps{
			Sound:            "default",
			ContentAvailable: true,
			MutableContent:   true,
		},
	},
},
```

In the first question listed by @darshankawar , it was suggested to add these `.Aps` fields, and use the APNSPayload field `Headers` , and provide it a priority `5` and type `background`, which got background notifications working but broke foreground notifications. I then just added/removed stuff until it worked for both. I am 100% sure I tested this at some point, but I think this identified an issue in the development process, the way we are using local kubernetes and building images for it, and they are not updating the image used for the running container under the right circumstances.

I'll close the issue, thanks for your help
